core/deadline.go

```diff
@@ -38,9 +38,9 @@ type slotTimeProvider interface {
 // may only be used by a single goroutine. So, multiple instances are required
 // for different components and use cases.
 type Deadliner interface {
-	// Add adds a duty to be notified of the Deadline via C.
+	// Add adds a duty to be notified of the Deadline via C. It returns true if the duty is added successfully.
```

Comment:
Suggest: `Add returns true if the duty was added for future deadline scheduling. It is idempotent and returns true if the duty was previously added and still awaits deadline scheduling. It returns false if the duty has already expired and cannot therefore be added for scheduling.`